### PR TITLE
Remove `objectWithout` usage from `elements` package

### PR DESCRIPTION
**Pull request by swissspidy**
## Context

I just noticed this is the only reason for the dependency on the `design-system` package. That util is not used anywhere else (the `story-editor` has its own copy).

## Summary

Remove `objectWithout` usage from `elements` package, removing the `design-system` package.

Fixes a bug in `duplicatePage` too, I think..?

Also removes a similar usage from the `patterns` package that I stumbled upon, making the code a bit more readable

## Testing Instructions

- [X] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #

### CODE REVIEW

File: packages/elements/src/utils/duplicatePage.js [hunk 1]
```diff
@@ -25,18 +21,15 @@ import createNewElement from './createNewElement';
 import duplicateElement from './duplicateElement';
 
 const duplicatePage = (oldPage) => {
-  // Remove title and postId for inserting the page.
+  // Remove title and templateId for inserting the page.
```
Comment (miina) [inline packages/elements/src/utils/duplicatePage.js]: I tried to test this part and can't see when would a template / post ID be assigned at all. Thought that perhaps when  inserting a page template and then duplicating it but doesn't seem like it. Were you able to test in which case we need to remove the title and ID when duplicating?

Comment (swissspidy) [reply packages/elements/src/utils/duplicatePage.js]: Indeed, doesn't look like that happens at all.

I tried inserting default page templates and custom saved page templates, but there's no `templateId` leaking into this.

Looking at `handlePageClick`, it's specifically omitted:

https://github.com/GoogleForCreators/web-stories-wp/blob/7f3af11a2fc2bc0e35733d1e79462f39918b2101/packages/story-editor/src/components/library/panes/pageTemplates/templateList.js#L72-L76

So yeah, seems like it's safe to remove this comment & check.

For context, originally introduce in #7290 and then adjusted in #7318

**Review by miina — APPROVED** [review]

**Review by spacedmonkey — APPROVED** [review]

File: packages/elements/src/utils/createNewElement.js [hunk 2]
```diff
@@ -37,9 +36,11 @@ const createNewElement = (type, attributes = {}) => {
     type,
     id: uuidv4(),
   };
+
   // There's an exception for the background shape that should not get all the default attributes.
   if (attributes.isDefaultBackground) {
-    return objectWithout(newElement, ['backgroundColor']);
+    const { backgroundColor: _, ...newElementWithoutBgColor } = newElement;
```
Comment (barklund) [inline packages/elements/src/utils/createNewElement.js:42]: Why rename it to `_` if unused anyway?

**Review by barklund — APPROVED** [review]
Looks good, makes sense to use the advanced language features.

File: packages/elements/src/utils/createNewElement.js (hunk 2 above)
Comment (swissspidy) [reply packages/elements/src/utils/createNewElement.js:42]: ESLint complained otherwise, surprisingly

**Review by timarney — APPROVED** [review]

**Pull request merged**